crates/executors/src/command.rs

```diff
@@ -1,6 +1,52 @@
+use std::path::PathBuf;
+
 use schemars::JsonSchema;
 use serde::{Deserialize, Serialize};
+use shlex::Quoter;
+use thiserror::Error;
 use ts_rs::TS;
+use workspace_utils::shell::resolve_executable_path;
+
+use crate::executors::ExecutorError;
+
+#[derive(Debug, Error)]
+pub enum CommandBuildError {
+    #[error("base command cannot be parsed: {0}")]
+    InvalidBase(String),
+    #[error("base command is empty after parsing")]
+    EmptyCommand,
+    #[error("failed to quote command: {0}")]
+    QuoteError(#[from] shlex::QuoteError),
+}
+
+#[derive(Debug, Clone)]
+pub struct CommandParts {
+    program: String,
+    args: Vec<String>,
+}
+
+impl CommandParts {
+    pub fn new(program: String, args: Vec<String>) -> Self {
+        Self { program, args }
+    }
+
+    pub fn into_resolved(self) -> Result<(PathBuf, Vec<String>), ExecutorError> {
+        let CommandParts { program, args } = self;
+        let executable = resolve_executable_path(&program)
+            .ok_or_else(|| ExecutorError::ExecutableNotFound { program })?;
+        Ok((executable, args))
+    }
+
+    pub fn to_shell_string(&self) -> Result<String, CommandBuildError> {
```

Comment:
We can remove this and associated logic now that the watchkill script has been removed